docs/en/02_Developer_Guides/15_Customising_the_Admin_Interface/07_ReactJS_Redux_and_GraphQL.md

```diff
@@ -0,0 +1,666 @@
+	// Applies to all text fields in AssetAdmin
+	updater.react('TextField.AssetAdmin', MyTextField);
+
+	// Applies to all text fields in AssetAdmin editform
+	updater.react('TextField.AssetAdmin.FileEditForm', MyTextField);
+
+	// Applies to any textfield named "Title" in AssetAdmin
+	updater.react('TextField.AssetAdmin.*.Title', MyTextField);
+
+	// Applies to any textfield named "Title" in any admin
+	updater.react('TextField.*.*.Title', MyTextField);
+
+	// Applies to all text fields in AssetAdmin
+	updater.react('TextField.AssetAdmin', MyTextField);
+
+})
+```
+
+ To apply context-based transformations, you'll need to know the context of the component you want to customise. To learn this,
+ open your React Developer Tools (see above) window and inspect the component name. The
+ context of the component is displayed between two square brackets, appended to the original name, for example:
+ `TextField[TextField.AssetAdmin.FileEditForm.Title]`. The context description is hierarchical, starting
+ with the most general category (in this case, "Admin") and working its way down to the most specific
+ category (Name = 'Title'). You can use Injector to hook into the level of specificity that you want.
+
+
+# Customising React components with Injector
+
+When middleware is used to customise a React component, it is known as a [higher order component](https://facebook.github.io/react/docs/higher-order-components.html).
+
+Using the `PhotoItem` example above, let's create a customised `PhotoItem` that allows a badge, perhaps indicating that it is new to the gallery.
+
+```js
+const enhancedPhoto = (PhotoItem) => (props) => {
+	const badge = props.isNew ? 
+	  <div className="badge">New!</div> : 
+	  null;
+
+	return (
+		<div>
+			{badge}
+			<PhotoItem {...props} />
+		</div>
+	);
+}
+
+const EnhancedPhotoItem = enhancedPhoto(PhotoItem);
+
+<EnhancedPhotoItem isNew={true} size={300} />
+```
+
+Alternatively, this component could be expressed with an ES6 class, rather than a simple
+function.
+
+```js
+const enhancedPhoto = (PhotoItem) => {
+	return class EnhancedPhotoItem extends React.Component {
+		render() {
+			const badge = this.props.isNew ? 
+			  <div className="badge">New!</div> : 
+			  null;
+
+			return (
+				<div>
+					{badge}
+					<PhotoItem {...this.props} />
+				</div>
+			);
+
+		}
+	}
+}
+```
+
+When components are stateless, using a simple function in lieu of a class is recommended.
+
+
+## Using dependencies within your React component
+
+If your component has dependencies, you can add them via the injector using the `inject()`
+higher order component. The function accepts the following arguments:
+
+```js
+inject(Component, [dependencies], mapDependenciesToProps)
+```
+* **Component** The component definition to inject into
+* **[dependencies]**: An array of dependencies (or a string, if just one)
+* **mapDependenciesToProps**: (optional) All depdencies are passed into this function as params. The function
```

Comment:
Have you checked if ES6 has a way to get argument names, which would save us from repeating those? I assume that's not possible though